insights/parsers/pcs_config.py

```diff
@@ -0,0 +1,243 @@
+    ...  Meta Attrs: interleave=true ordered=true
+    ...  Resource: res-1 (class=ocf provider=pacemaker type=controld)
+    ...  Operations: start interval=0s timeout=90 (dlm-start-interval-0s)
+    ...              stop interval=0s timeout=100 (dlm-stop-interval-0s)
+    ...              monitor interval=30s on-fail=fence (dlm-monitor-interval-30s)
+    ...
+    ... Stonith Devices:
+    ... Fencing Levels:
+    ...
+    ... Location Constraints:
+    ...  Resource: fence-1
+    ...  Resource: res-1
+    ... Ordering Constraints:
+    ... Colocation Constraints:
+    ...
+    ... Resources Defaults:
+    ...  No defaults set
+    ... Operations Defaults:
+    ...  No defaults set
+    ...
+    ... Cluster Properties:
+    ...  cluster-infrastructure: corosync
+    ...  cluster-name: cluster-1
+    ... '''.strip()
+    >>>
+    >>> shared = {PCSConfig: PCSConfig(context_wrap(pcs_config_output))}
+    >>> pcs_config = shared[PCSConfig]
+    >>> pcs_config.get("Corosync Nodes")
+    ['node-1', 'node-2']
+    >>> pcs_config.get("Cluster Properties")
+    ['cluster-infrastructure: corosync', 'cluster-name: cluster-1']
+    >>> pcs_config.get_resources_clones()
+    {'clone-1': ['Meta Attrs: interleave=true ordered=true', 'Resource: res-1 (class=ocf provider=pacemaker type=controld)', 'Operations: start interval=0s timeout=90 (dlm-start-interval-0s)', 'stop interval=0s timeout=100 (dlm-stop-interval-0s)', 'monitor interval=30s on-fail=fence (dlm-monitor-interval-30s)', '']}
+
+"""
+
+
+from .. import parser, CommandParser
+from insights.specs import Specs
+
+
+@parser(Specs.pcs_config)
+class PCSConfig(CommandParser):
+    """Class to process the output of ``pcs config`` command.
+
+    Attributes:
+        data (dict): A dictionary containing all line. Keys sorted based on keywords of the output.
+            the form::
+            {
+                "Cluster Name": "cluster-1",
+                "Corosync Nodes": [
+                    "node-1",
+                    "node-2"
+                ],
+                "Pacemaker Nodes": [
+                    "node-1",
+                    "node-2"
+                ],
+                "Resources": [
+                    "Clone: clone-1",
+                    "Meta Attrs: interleave=true ordered=true",
+                    "Resource: res-1 (class=ocf provider=pacemaker type=controld)",
+                    "Operations: start interval=0s timeout=90 (dlm-start-interval-0s)",
+                    "stop interval=0s timeout=100 (dlm-stop-interval-0s)",
+                    "monitor interval=30s on-fail=fence (dlm-monitor-interval-30s)"
+                ],
+                "Cluster Properties": [
+                    "cluster-infrastructure: corosync",
+                    "cluster-name: cluster-1"
+                ]
+            }
+    Methods:
+        get_resources_clones(self) (dict): Returns a dictionary of ``Clones`` in ``Resources`` keyed with their name.
+            the form::
+            {
```

Comment:
There should be a blank line after `the form::` and the dictionary example should be indented 1 tab.  Same with the next method.